hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java

```diff
@@ -309,8 +309,16 @@ public CascadeStyle getCascadeStyle() {
 			}
 		};
 
-		final FetchTiming fetchTiming = bootProperty.isLazy() ? FetchTiming.DELAYED : FetchTiming.IMMEDIATE;
-		final FetchStyle fetchStyle = bootProperty.isLazy() ? FetchStyle.SELECT : FetchStyle.JOIN;
+		final FetchTiming fetchTiming;
+		final FetchStyle fetchStyle;
+		if ( declaringType instanceof EmbeddableMappingType ) {
+			fetchTiming = FetchTiming.IMMEDIATE;
+			fetchStyle = FetchStyle.JOIN;
```

Comment:
IMO we should emit a warning about this in case fetch timing was `DELAYED`. Are we sure that we must use `JOIN` fetching?

Reply:
I don't think the JOIN really matters here.  But then again, I don't think it matters what the value is so... :)

I guess I'd argue... "what's the other option?"?  SELECT?

Reply:
Also, how would you even denote an embeddable/embedded is DELAYED?  I don't think there is a way

Reply:
Ah, sorry... this is about LAZY for an attribute *on* the embeddable, not the embedded itself.  I misunderstood.

Then I agree that a warning would be nice